src/components/XdrViewer.js

```diff
@@ -18,15 +27,17 @@ function XdrViewer(props) {
 
   let xdrTypeIsValid = _.indexOf(xdrTypes, state.type) >= 0;
   let treeView, errorMessage;
-  if (state.input === '') { 
+  if (state.input === '') {
     errorMessage = <p>Enter a base-64 encoded XDR blob to decode.</p>;
   } else if (!xdrTypeIsValid) {
     errorMessage = <p>Please select a XDR type</p>;
   } else {
     try {
       treeView = <TreeView nodes={extrapolateFromXdr(state.input, state.type)} fetchedSigners={state.fetchedSigners} />
+      tLogEvent(metricsEvents.decodeSuccess, {type: state.type})
     } catch (e) {
       console.error(e)
+      tLogEvent(metricsEvents.decodeFailed, {type: state.type})
```

Comment:
This and [L37](https://github.com/stellar/laboratory/pull/370/files#diff-ca1739756f192224e8ee7af3c7dfce76R37) are debounced metrics loggers run within the render function of this component, which isn't ideal. Pretty much the only useful metric to collect here, though, is whether the XDR was successfully decoded, and that doesn't happen in response to a Redux action.

Reply:
Seems like this should then use `componentDidUpdate` instead?

Reply:
That'd be a much more significant refactor, since this is a function component and everything is calculated there. Would have to add additional information as well to tell whether it succeeded or failed, too.